starboard/shared/starboard/media/codec_util.cc

```diff
@@ -107,13 +108,101 @@ SbMediaAudioCodec GetAudioCodecFromString(const char* codec,
     return kSbMediaAudioCodecPcm;
   }
 #if SB_API_VERSION >= 15
-  if (strcmp(codec, "iamf") == 0 || strncmp(codec, "iamf.", 5) == 0) {
+  if (strcmp(codec, "iamf") == 0 || IsIamfMimeType(codec)) {
     return kSbMediaAudioCodecIamf;
   }
 #endif  // SB_API_VERSION >= 15
   return kSbMediaAudioCodecNone;
 }
 
+#if SB_API_VERSION >= 15
+bool IsIamfMimeType(std::string mime_type) {
+  // Reference: Immersive Audio Model and Formats;
+  //            v1.0.0
+  //            6.3. Codecs Parameter String
+  // (https://aomediacodec.github.io/iamf/v1.0.0-errata.html#codecsparameter)
+  if (mime_type.find("iamf") == std::string::npos) {
+    return false;
+  }
+
+  // 4   FOURCC string "iamf".
+  // +1  delimiting period.
+  // +3  primary_profile as 3 digit string.
+  // +1  delimiting period.
+  // +3  additional_profile as 3 digit string.
+  // +1  delimiting period.
+  // +9  The remaining string is one of "Opus", "mp4a.40.2", "fLaC", or "ipcm".
+  constexpr int kMaxIamfCodecIdLength = 22;
+
+  if (mime_type.size() > kMaxIamfCodecIdLength) {
+    return false;
+  }
+
+  std::vector<std::string> vec = SplitString(mime_type, '.');
+  if (vec.size() != 4 && vec.size() != 6) {
+    return false;
+  }
+
+  if (vec[0] != "iamf") {
+    return false;
+  }
+
+  // The primary profile string should be three digits, and should be between 0
+  // and 255 inclusive.
+  int primary_profile;
+  std::stringstream stream(vec[1]);
```

Comment:
vec[1] is std::string, should we consider to use `std::stoi()` to convert string to integer here?